lib/thrift/parser/models.ex

```diff
@@ -254,6 +254,31 @@ defmodule Thrift.Parser.Models do
 
       %Union{name: name, fields: fields}
     end
+
+    def validator(%Union{}, var_name) do
+      union_var = Macro.var(var_name, nil)
+      quote do
+        set_fields = unquote(union_var)
+        |> Map.delete(:__struct__)
+        |> Enum.reject(fn {_, val} -> is_nil(val) end)
+        case set_fields do
+          [] ->
+            :ok
+
+          [_] ->
+            :ok
+
+          set_fields ->
+            raise Thrift.Union.TooManyFieldsSetException.new(set_fields)
+        end
+      end
+    end
```

Comment:
This is code generation stuff and not related to parsing. It would be more at home somewhere under Thrift.Generator than Thrift.Parser. Maybe we should generate a `valid?` method on union models that does this check.

I'm concerned about the performance impact of doing a Map.delete and Enum.reject every time we serialize a union. I suspect it would be significantly faster to use the match on the map itself to enforce that only one field is set. We're already doing that match anyway.

```elixir
union MyUnion {1: string foo, 2: string bar, 3: string baz}

def serialize(%MyUnion{foo: nil, bar: nil, baz: nil})
def serialize(%MyUnion{foo: foo, bar: nil, baz: nil})
def serialize(%MyUnion{foo: nil, bar: bar, baz: nil})
def serialize(%MyUnion{foo: nil, bar: nil, baz: baz})
```

If you submit this PR with the current approach, would you mind if I explored such a refactoring in a subsequent PR?